### ISSUE TITLE
The "Overall debt:" time estimate is no longer displayed anywhere

### ISSUE THREAD

**Issue by ozmium** [detekt/detekt#5500]
## Expected Behavior
When I run detekt as a gradle task, it should display the `"Overall debt"` time estimate in the console, and also in the HTML report somewhere -- maybe near the top? Also, the time estimates of the groups that make up the total time should be displayed in the HTML report, as well. For example:

```
HTML report

Metrics
======
* x number of properties

Complexity Report
=============
* x lines of code (loc)

Findings
========
* Total: 12
* Overall debt: 1 hour 30 minutes debt

* complexity: 2 (1 hour debt)
* ...
* style: 10 (30 minutes debt)
* ...
```

In the past in older versions, like `1.15.0`, the `"Overall debt:"` would be displayed on the console. But now with recent versions like `1.20.0`, it doesn't display anywhere.

The code to show "Overall debt" seems to be present in the latest version (19 June 2022) of `Reporting.kt`: https://github.com/detekt/detekt/blob/06ee2394410256a7143bb07cdcfac52f8d76c5c0/detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/reporting/Reporting.kt#L41

It's also mentioned here: https://github.com/detekt/detekt/issues/786

## Observed Behavior
When I run detekt as a gradle task, the "Overall debt" time estimate is no longer displayed anywhere.

## Steps to Reproduce

1. Use Android Studio to create an Android app. Any sample code is OK, as long as it has some Kotlin files.
2. Deliberately add some bad things in the code that Detekt shoudl report on. For example, very long lines over 120 characters, empty functions, throw RuntimeException(), no new lines at end of the file, unused private class fields, and so on.
3. Add detekt version 1.20.0 as a gradle dependency to the project.
4. Optional: You may need to tell detekt not to fail the gradle task if too many warnings are detekted. Something like this in `app/build.gradle`:

```gradle
detekt {
    toolVersion = "$detekt_version"
    // Do not fail the build if the maxIssues count was reached.
    ignoreFailures = true

    reports {
        html.enabled = true
        xml.enabled = false
        txt.enabled = false
    }

    println("Detekt is configured")
}
```
 
5. Run `./gradlew detekt` or `./gradlew detektDebugAndroidTest` or `./gradlew detektDebugUnitTest`
6. Observe the output on the command line and also in the HTML report file in `$buildDir/reports/detekt/detekt.html`

## Context

How has this issue affected me:
* On the older versions of Detekt, the debt time estimates were useful to get a summary or a rough estimate of the amount of work and time required to fix all the reported issues. The newer versions no longer show this useful statistic.

## Your Environment

* Version of detekt used: 1.20.0
* Version of Gradle used (if applicable):
* * `gradle-wrapper.properties`: distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
* * `build.gradle` -> buildscript { dependencies { classpath 'com.android.tools.build:gradle:7.2.2' ... } }
* Gradle scan link (add `--scan` option when running the gradle task): Not available
* Operating System and version: Windows 11/10/8, macOS 12 Monterey or 11 Big Sur or 10.15 Catalina, Ubuntu Linux 18.04 - 22.10
* Link to your project (if it's a public repository): Not available

**Comment by schalkms** [comment]
How does your `detekt-config.yml` look like?
Is `FindingsReport` or `FileBasedFindingsReport` *not* excluded?
Both do actually report the overall debt, exactly what you want to see.

https://github.com/detekt/detekt/blob/621bcc912114f22d93c300ed1ec9f40fa919406b/detekt-core/src/main/resources/default-detekt-config.yml#L34

Please note that we made some changes to uniform the behavior.

**Comment by ozmium** [comment]
@schalkms This is my `detekt-config.yml`:

```yml
build:
  maxIssues: 10

style:
  ForbiddenComment:
    values: 'FIXME:,STOPSHIP:'

  MaxLineLength:
    maxLineLength: 120

  MagicNumber:
    active: false

complexity:
  TooManyFunctions:
    thresholdInClasses: 20
    thresholdInObjects: 20
```

`app/build.gradle` has:

```gradle
detekt {
    toolVersion = "$detekt_version" // "1.20.0"
    buildUponDefaultConfig = true
    config = files("$projectDir/detekt-config.yml")
    ignoreFailures = true

    reports {
        html.enabled = true
        xml.enabled = false
        txt.enabled = false
    }

    println("Detekt is configured") // report created in $buildDir/reports/detekt/detekt.html
}
```

But even when I comment out these lines in `app/build.gradle` to not use the `detekt-config.yml` file ...

```gradle
//buildUponDefaultConfig = true // disable
//config = files("$projectDir/detekt-config.yml") // disable
```

... the same problem still happens and the "Overall debt" is not displayed anywhere.

Please advise on how to setup the `detekt-config.yml` or any other file, to enable the "Overall debt" to appear.

**Comment by schalkms** [comment]
I'm afraid I don't see where you put the `console-report` section in your `detekt-config.yml` file.

This needs to be added. Otherwise the console-reports are excluded by default, as you can see in the linked default detekt config. Detekt runs *silent*  by default at least in the console. You need to comment the report you would like to have included with a leading `#`.

https://github.com/detekt/detekt/blob/621bcc912114f22d93c300ed1ec9f40fa919406b/detekt-core/src/main/resources/default-detekt-config.yml#L34-L42

**Comment by schalkms** [comment]
The relevant section in the docs can be found here:
https://detekt.dev/docs/introduction/configurations#console-reports

**Comment by BraisGabin** [comment]
I see two different issues here. One is fixed with what @schalkms said. We changed our default console report in a recent version of detekt. If you like the old one you just need to change your confiugration to this one:

```yaml
 console-reports: 
   active: true 
   exclude: 
      - 'ProjectStatisticsReport' 
      - 'ComplexityReport' 
      - 'NotificationReport' 
   #   - 'FindingsReport' 
      - 'FileBasedFindingsReport' 
      - 'LiteFindingsReport' 
```

The other issue is to add the debt time in the `html` report. That seems like a good idea to me but I would ask you to create a new issue to only track that. This way we could add the `help wanted` tag and anyone can grab it.

I'm closing this one because I think that the issue/question is solved. If that's not the case, please comment again :)

**Comment by ozmium** [comment]
Thanks for letting me know how to configure it. I finally managed to get it working with this configuration:

**detekt-config.yml**

```yml
console-reports:
  active: true
  exclude:
    - 'ProjectStatisticsReport'
    - 'ComplexityReport'
    - 'NotificationReport'
    - 'FindingsReport'
  #  - 'FileBasedFindingsReport'
    - 'LiteFindingsReport'
```

However, there are some problems I noticed along the way:

* It's very difficult to understand how to configure things like this on my own, or by looking through the documentation.  Even doing a search in the GitHub Issues was not enough, since I couldn't find any pre-existing reports.
* Having to comment things out to switch them on is really odd to me. The logic is confusing. Isn't it easier to invert the logic, to make it an "include:" group, and then add things to switch on?
* In the given documentation, https://detekt.dev/docs/introduction/configurations#console-reports , there is no mention of `"Overall debt"`, `"time estimate"`, `"debt"`, `"hours"`, `"days"` or `"time"`. All these search terms give zero results. How would anyone know how to switch on "Overall debt", when it's not written in the documentation anywhere?
* Some of the terms in the documentation have unclear or missing explanations. For example, what is a `"LiteFindingsReport"`? What is a `"FileBasedFindingsReport"` and why it is in the `"console-reports"` section? I could only figure it out through trial and error.
* There is no migration guide, for how to migrate from old versions of detekt to newer versions. For example, it would be nice to have an explanation of which things are no longer shown in the newer versions, with a side by side comparison of the old way to enable it with the new way.
* As mentioned by @BraisGabin , I could not get the "Overall debt" to display in the HTML reports, as well as the time estimate for each group in the HTML report. It only displays on the console for now.

I would be happy to try to help out by adding some more documentation. But I don't know if it's really wanted by the members.

**Comment by BraisGabin** [comment]
Sure it is! Any improvement to our documentation is more than welcome!

On our changelog we have information about that change. We said that we changed the default console report.

About the exclude/include you are completely right. We hope to change it for 2.0

**Comment by ozmium** [comment]
Another problem I've just noticed, when using this:

### detekt-config.yml

```yml
console-reports:
  active: true
  exclude:
    - 'ProjectStatisticsReport'
    - 'ComplexityReport'
    - 'NotificationReport'
    - 'FindingsReport'
  #  - 'FileBasedFindingsReport'
    - 'LiteFindingsReport'
```

The "Overall debt" is only shown **once** when there is a code change. If you run the command again, without changing any code, it will not show the "overall debt" time estimate in the console. I don't understand why it works like that or how that benefits anyone, but I would prefer if it always showed the "overall debt" every time the command is run.

**Comment by cortinico** [comment]
> The "Overall debt" is only shown once when there is a code change. If you run the command again, without changing any code, it will not show the "overall debt" time estimate in the console. 

Please open a separate issue.
I believe your problem is related to Gradle and UP-TO-DATE-ness. You can avoid it by invoking your task with `--rerun-tasks` (or `--rerun` in the latest Gradle versions).
